### PR TITLE
Fix/time computing

### PR DESCRIPTION
**Pull request by ncovic1**

### CODE REVIEW

File: src/configurations/DRGBTConfig.cpp [hunk 1]
```diff
@@ -4,15 +4,16 @@
 
 #include "DRGBTConfig.h"
 
-unsigned long DRGBTConfig::MAX_NUM_ITER             = 10e6;
-int DRGBTConfig::MAX_ITER_TIME                      = 100;
-int DRGBTConfig::MAX_PLANNING_TIME                  = 60e3;
+unsigned long DRGBTConfig::MAX_NUM_ITER             = 1e9;
```
Comment (dinkoosmankovic) [inline src/configurations/DRGBTConfig.cpp]: size_t

File: src/configurations/DRGBTConfig.cpp [hunk 2]
```diff
@@ -4,15 +4,16 @@
...
+float DRGBTConfig::MAX_ITER_TIME                    = 0.050;
+float DRGBTConfig::MAX_PLANNING_TIME                = 60;
 int DRGBTConfig::INIT_HORIZON_SIZE                  = 10;
```
Comment (dinkoosmankovic) [inline src/configurations/DRGBTConfig.cpp]: Change obviously positive integer types to size_t

File: src/planners/PlannerInfo.cpp [hunk 3]
```diff
@@ -29,10 +29,10 @@ void PlannerInfo::addCostConvergence(const std::vector<float> &cost_convergence)
 		PlannerInfo::cost_convergence.emplace_back(cost_convergence[i]);
 }
 
-void PlannerInfo::addRoutineTime(int time, int idx)
+void PlannerInfo::addRoutineTime(float time, int idx)
```
Comment (dinkoosmankovic) [inline src/planners/PlannerInfo.cpp:32]: size_t

Fix the vector size before the for loop.

File: src/planners/rbt/RGBTConnect.cpp [hunk 4]
```diff
@@ -97,20 +97,20 @@ std::tuple<base::State::Status, std::shared_ptr<std::vector<std::shared_ptr<base
 {
     float d_c = ss->computeDistance(q);
 	std::shared_ptr<base::State> q_new = q;
-	std::shared_ptr<std::vector<std::shared_ptr<base::State>>> q_new_list = std::make_shared<std::vector<std::shared_ptr<base::State>>>();
+	std::vector<std::shared_ptr<base::State>> q_new_list;
```
Comment (dinkoosmankovic) [inline src/planners/rbt/RGBTConnect.cpp:100]: Fix the size.

File: src/planners/trajectory/Spline5.cpp [hunk 5]
```diff
@@ -255,6 +255,38 @@ bool planning::trajectory::Spline5::checkConstraints(int idx, float t_f)
...
+    return solveQubicEquation(20*a(idx), 12*b(idx), 6*c(idx), 2*d(idx));
+}
+
+std::vector<float> planning::trajectory::Spline5::getMaxAccelerationTimes(int idx)
+{
+    std::vector<float> t_max;
+
+    if (a(idx) != 0)
+    {
+        float D = 576*b(idx)*b(idx) - 1440*a(idx)*c(idx);
```
Comment (dinkoosmankovic) [inline src/planners/trajectory/Spline5.cpp:269]: All magic numbers should be declared as constants.

File: src/planners/PlannerInfo.cpp (hunk 3 above)
Comment (ncovic1) [reply src/planners/PlannerInfo.cpp:32]: Hmm, unfortunately, I don't know how many elements will be added to the vector.

File: src/planners/rbt/RGBTConnect.cpp (hunk 4 above)
Comment (ncovic1) [reply src/planners/rbt/RGBTConnect.cpp:100]: The same reason, I don't know how many elements will be added to the vector.

File: src/planners/trajectory/Spline5.cpp (hunk 5 above)
Comment (ncovic1) [reply src/planners/trajectory/Spline5.cpp:269]: These magic numbers are very specific for each type of spline, e.g., these ones are for quintic spline (Spline5), and they will never be changed, and they cannot be changed. On the other hand, after we implement Spline3 or Spline4, they will have their unique magic numbers. However, there are too many magic numbers in Spline5, thus I don't know how to name all of them. :D

File: src/configurations/DRGBTConfig.cpp (hunk 2 above)
Comment (ncovic1) [reply src/configurations/DRGBTConfig.cpp]: OK, tnx.

File: src/configurations/DRGBTConfig.cpp (hunk 1 above)
Comment (ncovic1) [reply src/configurations/DRGBTConfig.cpp]: OK, tnx.

File: src/planners/rbt/RGBTConnect.cpp (hunk 4 above)
Comment (ncovic1) [reply src/planners/rbt/RGBTConnect.cpp:100]: This has been already discussed in #13.

**Review by dinkoosmankovic — APPROVED** [review]

**Pull request merged**